internal/pkg/flink/internal/store/store_utils.go

```diff
@@ -110,6 +110,24 @@ func (s *Store) processUseStatement(statement string) (*types.ProcessedStatement
 	}
 
 	s.Properties[configKey] = configVal
+	if s.appOptions.GetContext() != nil {
+		if configKey == config.ConfigKeyCatalog {
+			err := s.appOptions.Context.SetCurrentFlinkCatalog(configVal)
+			if err != nil {
+				return nil, &types.StatementError{Message: err.Error()}
```

Comment:
Every error in this function is wrapped in the same way. Can we wrap it outside of the function, or remove duplication some other way?

Reply:
factoring out the error wrap won't dry it up - extra complexity from having a method that literally wraps the repeating return line + it's going to make it harder to read (when you look at statement error now you need to do a jump to find out what the message is) I'm inclined not to touch it for now